app/frontend/courses/curricula/components/CoursesCurriculum__SubmissionComments.res

```diff
@@ -0,0 +1,124 @@
+%%raw(`import "./CoursesCurriculum__SubmissionComments.css";`)
+let str = React.string
+let tr = I18n.t(~scope="components.CoursesCurriculum__SubmissionComments")
+
+open CoursesCurriculum__Types
+module CreateSubmissionCommentMutation = %graphql(`
+   mutation CreateSubmissionCommentMutation($comment: String!, $submissionId: String!) {
+     createSubmissionComment(comment: $comment, submissionId: $submissionId ) {
+       comment {
+         id
+         userId
+         comment
+         submissionId
+         user {
+            id
+            name
+            title
+            avatarUrl
+         },
+         createdAt
+         reactions {
+            id,
+            reactionableId,
+            reactionValue,
+            reactionableType,
+            userName,
+            updatedAt
+          }
+          moderationReports {
+            id,
+            userId,
+            reportableId,
+            reason,
+            reportableType
+          },
+       }
+     }
+   }
+   `)
+
+let toggleComments = (setShowComments, event) => {
+  ReactEvent.Mouse.preventDefault(event)
+  setShowComments(prevState => !prevState)
+}
+
+@react.component
+let make = (~currentUser, ~submissionId, ~comments) => {
+  let (submissionComments, setSubmissionComments) = React.useState(() => comments)
+  let (showComments, setShowComments) = React.useState(() => false)
+  let (newComment, setNewComment) = React.useState(() => "")
+
+  let handleInputChange = event => {
+    setNewComment(ReactEvent.Form.currentTarget(event)["value"])
+  }
+
+  let handleCreateSubmissionComment = event => {
+    ReactEvent.Mouse.preventDefault(event)
+    CreateSubmissionCommentMutation.make({comment: newComment, submissionId})
+    |> Js.Promise.then_(response => {
+      switch response["createSubmissionComment"]["comment"] {
+      | Some(createdComment) =>
+        setNewComment(_ => "")
+        setSubmissionComments(existingComments =>
+          Js.Array2.concat([createdComment->Comment.decode], existingComments)
+        )
+      | None => ()
+      }
+      Js.Promise.resolve()
+    })
+    |> ignore
+  }
+
+  <div className="w-full">
+    <div className="flex items-center justify-between mx-auto">
+      <div id={"show_comments-" ++ submissionId}>
+        <button
```

Comment:
- I'm not a huge fan of how this button needs to be clicked to show comments.
- This button _needs_ to show the number of comments on a submission, if there are any. i.e., if there are no comments, it should probably read _Add a comment_; otherwise, it should say _X comments_

Reply:
@harigopal Yes. Me and Vinu did have a discussion on this, and we were thinking we would improve this on further iterations as people start using it. So we kept a basic comment toggle button that didn't change dynamically. 

But I guess it's too basic 😄. Made the change you suggested with one variation after discussion with Vinu. If there are no comments, then there is no extra toggle comment button. You can directly see the "Write you comment" input.